v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/cassandra/SourceColumn.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra;
+
+import com.google.auto.value.AutoValue;
+import java.io.Serializable;
+
+/**
+ * Represents a source column schema with type, primary key information, and column kind. This class
+ * provides immutable representation of a source column with necessary metadata.
+ */
+@AutoValue
+public abstract class SourceColumn implements Serializable {
```

Comment:
"Column" is used to denote the columns of table throughout the code. Here is seems to be referring to other schema elements. 

I suggest
1. Remove the name sourceColumn and call it something more representative of what it holds.
2. I'm not sure from this PR how this fits with the session file representation of items present in the package.

Reply:
@bharadwaj-aditya we are using this same as DDL Object where we are populating Table and Column based on Resultset so I guess The naming is aline with Requirement as we are already used for Spanner DDL object